From 6c22458690b76ca57e56655fed5a843847405ab3 Mon Sep 17 00:00:00 2001
From: Thomas Gleixner <tglx@linutronix.de>
Date: Mon, 17 Apr 2017 09:57:10 +0200
Subject: [PATCH 348/360] x86/intel_rdt: Get rid of anon union

commit a83827d04f081b1a73845553c7f60ab9cfd13767 upstream.

gcc-4.4.3 fails to statically initialize members of a anon union.
See: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=10676

The storage saving is not really worth it and aside of that it will catch
usage of the cache member for bandwidth and vice versa easier.

Fixes: 05b93417ce5b ("x86/intel_rdt/mba: Add primary support for Memory Bandwidth Allocation (MBA)")
Reported-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 arch/x86/include/asm/intel_rdt.h |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/x86/include/asm/intel_rdt.h b/arch/x86/include/asm/intel_rdt.h
index 507ed16..aea7ebb 100644
--- a/arch/x86/include/asm/intel_rdt.h
+++ b/arch/x86/include/asm/intel_rdt.h
@@ -166,10 +166,8 @@ struct rdt_resource {
 				 struct rdt_resource *r);
 	int			data_width;
 	struct list_head	domains;
-	union {
-		struct rdt_cache	cache;
-		struct rdt_membw	membw;
-	};
+	struct rdt_cache	cache;
+	struct rdt_membw	membw;
 	struct rftype		*info_files;
 	int			nr_info_files;
 	const char		*format_str;
-- 
1.7.5.4

